vnext/Microsoft.ReactNative/RedBox.cpp

```diff
@@ -11,117 +11,303 @@
 
 namespace Mso::React {
 
-std::function<void(facebook::react::JSExceptionInfo &&)> CreateRedBoxExceptionCallback(
-    std::shared_ptr<facebook::react::MessageQueueThread> uiMessageQueue,
-    std::function<void()> &&reloadCallback) noexcept {
-  return [ uiMessageQueue, reloadCallback2 = std::move(reloadCallback) ](
-      facebook::react::JSExceptionInfo && jsExceptionInfo) noexcept {
-    uiMessageQueue->runOnQueue([jsExceptionInfo, reloadCallback3 = std::move(reloadCallback2)]() {
-      auto popup = winrt::Windows::UI::Xaml::Controls::Primitives::Popup{};
-
-      const winrt::hstring xamlString =
-          L"<Grid Background='{ThemeResource ContentDialogBackground}'"
+struct RedBox : public std::enable_shared_from_this<RedBox> {
+  RedBox(
+      Mso::WeakPtr<IReactHost> weakReactHost,
+      std::function<void(uint32_t)> &&onClosedCallback,
+      const JSExceptionInfo &&exceptionInfo)
```

Comment:
>const JSExceptionInfo && [](start = 6, length = 24)

const && - it is not clear what is the intent here. For move semantic please remove const. For const &, please remove one &.
For the move semantic we need std::move below in the initialization list.